actors/miner/tests/miner_actor_test_bitfield_queue.rs

```diff
@@ -0,0 +1,273 @@
+
+    let epoch = ChainEpoch::from(42);
+
+    queue.add_to_queue_values(epoch, [1, 3].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch, [2, 4].to_vec()).unwrap();
+
+    let expected: HashMap<_, Vec<u64>> = HashMap::new();
+    let mut bq_expectation = BQExpectation { expected };
+    bq_expectation.add(epoch, [1, 2, 3, 4].to_vec()).equals(&mut queue);
+}
+
+#[test]
+fn adds_values_to_different_epochs() {
+    let h = ActorHarness::new(0);
+    let rt = h.new_runtime();
+    let mut queue = empty_bitfield_queue(&rt, TEST_AMT_BITWIDTH);
+
+    let epoch1 = ChainEpoch::from(42);
+    let epoch2 = ChainEpoch::from(93);
+
+    queue.add_to_queue_values(epoch1, [1, 3].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch2, [2, 4].to_vec()).unwrap();
+
+    let expected: HashMap<_, Vec<u64>> = HashMap::new();
+    let mut bq_expectation = BQExpectation { expected };
+    bq_expectation.add(epoch1, [1, 3].to_vec()).add(epoch2, [2, 4].to_vec()).equals(&mut queue);
+}
+
+#[test]
+fn pop_until_from_empty_queue_returns_empty_bitfield() {
+    let h = ActorHarness::new(0);
+    let rt = h.new_runtime();
+    let mut queue = empty_bitfield_queue(&rt, TEST_AMT_BITWIDTH);
+
+    let next = queue.pop_until(42).unwrap();
+
+    //no values are returned
+    let count = next.0.len();
+    assert_eq!(0, count);
+}
+
+#[test]
+fn pop_until_does_nothing_if_until_parameter_before_first_value() {
+    let h = ActorHarness::new(0);
+    let rt = h.new_runtime();
+    let mut queue = empty_bitfield_queue(&rt, TEST_AMT_BITWIDTH);
+
+    let epoch1 = ChainEpoch::from(42);
+    let epoch2 = ChainEpoch::from(93);
+
+    queue.add_to_queue_values(epoch1, [1, 3].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch2, [2, 4].to_vec()).unwrap();
+
+    let next = queue.pop_until(epoch1 - 1).unwrap();
+
+    // no values are returned
+    let count = next.0.len();
+    assert_eq!(0, count);
+
+    let expected: HashMap<_, Vec<u64>> = HashMap::new();
+    let mut bq_expectation = BQExpectation { expected };
+    // queue remains the same
+    bq_expectation.add(epoch1, [1, 3].to_vec()).add(epoch2, [2, 4].to_vec()).equals(&mut queue);
+}
+
+#[test]
+fn pop_until_removes_and_returns_entries_before_and_including_target_epoch() {
+    let h = ActorHarness::new(0);
+    let rt = h.new_runtime();
+    let mut queue = empty_bitfield_queue(&rt, TEST_AMT_BITWIDTH);
+
+    let epoch1 = ChainEpoch::from(42);
+    let epoch2 = ChainEpoch::from(93);
+    let epoch3 = ChainEpoch::from(94);
+    let epoch4 = ChainEpoch::from(204);
+
+    queue.add_to_queue_values(epoch1, [1, 3].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch2, [5].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch3, [6, 7, 8].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch4, [2, 4].to_vec()).unwrap();
+
+    let next = queue.pop_until(epoch2).unwrap();
+
+    // values from first two epochs are returned
+    assert_bitfield_equals(&next.0, &[1, 3, 5]);
+
```

Comment:
check second half of tuple

Reply:
Returned a `(next, modified)` tuple and checked `modified` value to match Go tests.